### ISSUE TITLE
Parsing Elements/Hit test

### ISSUE THREAD

**Issue by mdwem** [SVGKit/SVGKit#316]
My path is in the format
<path id=​"o1-200" d=​"M202,564 202,878 290,878 300,867 300,874 316,874 316,863 338,863 344,866 344,929 353,929 353,938 636,935 642,929 642,800 765,800 765,812 769,816 779,816 779,909 883,909 883,842 893,842 893,837 938,838 938,820 958,820 958,823 962,823 962,817 982,817 993,806 1010,806 1010,833 1015,838 1015,884 1062,884 1062,894 1065,894 1065,896 1096,896 1096,885 1135,885 1135,936 1121,950 1121,1055 1264,1055 1278,1041 1278,949 1265,936 1251,936 1251,823 1273,823 1276,826 1365,826 1365,662 1287,662 1284,665 1251,665 1251,545 1262,545 1262,439 1188,439 1188,385 1135,385 1135,409 1117,409 1117,545 1135,545 1135,591 1105,591 1105,605 1097,605 1097,623 1053,623 1035,601C1041,599 1049,594 1049,577L1049,458C1049,424 997,424 997,458L997,469 997,475 902,475 902,477 767,477 767,555 761,555 761,576 752,576 752,633 737,633 737,573 731,573 731,582 729,582 729,550 287,550 285,554 244,554 240,550 232,550 228,554 217,554 217,559 209,559 209,564 202,564Z" centerpoint=​"852.21,722.36">​</path>​

When i do a hit test i can successfully get the name of the hit layer as o1-200.
How do i get the the centerpoint=​"852.21,722.36" during a hit test on the layer.

I am using the demo project as reference for hit layer testing.

**Comment by adamgit** [comment]
what is centepoint?

**Comment by mdwem** [comment]
centre of the bounding box

**Comment by adamgit** [comment]
You can get the CALayer that was hit, and then check its .bounds, I think?

**Comment by mdwem** [comment]
Yes..however ,bounding box does not works well for all shapes,so i devised my own method for calculating the center  and adding in the svg tag.After that i am not able to get this tag property the way i get name like hitlayer.name

**Comment by adamgit** [comment]
All attributes on svg eleemnts should be visible. Have youchecked in the
debuger?

**Comment by mdwem** [comment]
Yes...i can see id ,path,d but not centerpoint and how will i append this property with the hitlayer.I can only get name

**Comment by adamgit** [comment]
Sorry, I don't understand. You have the full SVG DOM tree, so you can search that for your SVG element that has same identifier / id (can't use id in Apple projects, reserved keyword), and read the data from it directly.

**Comment by mdwem** [comment]
For example,in your demo code  
NSLog(@"svg touched layer %@ (%@): {%.1f, %.1f} {%.1f, %.1f}", hitLayer.name, [hitLayer class], lastTappedLayer.frame.origin.x, lastTappedLayer.frame.origin.y, lastTappedLayer.frame.size.width, lastTappedLayer.frame.size.height);

i get the name of the layer as " o1-200" ,how can i get the center point property

**Comment by adamgit** [comment]
As per previous comment: you have the entire SVG DOM. Search the DOM, find the SVG element (subclass of SVGElement) that has the same id as your layer-name.

That object will have all your attributes on it (its a subclass of Element and of Node, and both those classes have ways for you to read your attribute)
